### PR TITLE
Workaround for get-environment-information not providing class data f…

### PR DESCRIPTION
**Pull request by bkeifer**
…or each entry.

The get-environment-information RPC output corresponds to the "show chassis environment" CLI command.  The output from this command in a multi-chassis EX switch environment is as follows:

```
> show chassis environment
Class Item                           Status     Measurement
Power FPC 0 Power Supply 0           OK
      FPC 1 Power Supply 0           OK
      FPC 2 Power Supply 0           OK
Temp  FPC 0 CPU Sensor               OK         37 degrees C / 98 degrees F
      FPC 0 PSU Sensor               OK         30 degrees C / 86 degrees F
      FPC 1 CPU Sensor               OK         38 degrees C / 100 degrees F
      FPC 1 PSU Sensor               OK         30 degrees C / 86 degrees F
      FPC 2 CPU Sensor               OK         44 degrees C / 111 degrees F
      FPC 2 PSU Sensor               OK         35 degrees C / 95 degrees F
Fans  FPC 0 Fan Tray 0               OK         Spinning at normal speed
      FPC 0 Fan Tray 1               OK         Spinning at normal speed
      FPC 1 Fan Tray 0               OK         Spinning at normal speed
      FPC 1 Fan Tray 1               OK         Spinning at normal speed
      FPC 2 Fan Tray 0               OK         Spinning at normal speed
```

That's all well and good on the command line, but the RPC output seems to correspond to the CLI output a bit _too_ faithfully.  PrettyPrinting from within the napalm-junos module reveals the structure:

```
[   (   'FPC 0 Power Supply 0',
        [('status', 'OK'), ('class', 'Power'), ('temperature', None)]),
    (   'FPC 1 Power Supply 0',
        [('status', 'OK'), ('class', None), ('temperature', None)]),
    (   'FPC 2 Power Supply 0',
        [('status', 'OK'), ('class', None), ('temperature', None)]),
    (   'FPC 0 CPU Sensor',
        [('status', 'OK'), ('class', 'Temp'), ('temperature', 37)]),
    (   'FPC 0 PSU Sensor',
        [('status', 'OK'), ('class', 'Temp'), ('temperature', 29)]),
    (   'FPC 1 CPU Sensor',
        [('status', 'OK'), ('class', None), ('temperature', 38)]),
    (   'FPC 1 PSU Sensor',
        [('status', 'OK'), ('class', None), ('temperature', 30)]),
    (   'FPC 2 CPU Sensor',
        [('status', 'OK'), ('class', None), ('temperature', 43)]),
    (   'FPC 2 PSU Sensor',
        [('status', 'OK'), ('class', None), ('temperature', 35)]),
    (   'FPC 0 Fan Tray 0',
        [('status', 'OK'), ('class', 'Fans'), ('temperature', None)]),
    (   'FPC 0 Fan Tray 1',
        [('status', 'OK'), ('class', None), ('temperature', None)]),
    (   'FPC 1 Fan Tray 0',
        [('status', 'OK'), ('class', None), ('temperature', None)]),
    (   'FPC 1 Fan Tray 1',
        [('status', 'OK'), ('class', None), ('temperature', None)]),
    (   'FPC 2 Fan Tray 0',
        [('status', 'OK'), ('class', None), ('temperature', None)])]
```

Note that each power supply, temperature sensor, or fan after those in the first chassis are given a 'class' of None, likely taken straight from the human-readable output of the CLI command.

This patch modifies the behavior of the main loop within get_environment() so that if a sensor object is encountered with a 'class' of None, the previous valid class is used.

This works on all models of EX switch that I have access to test against.  Testing against other JunOS platforms is likely warranted.

### CODE REVIEW

File: napalm_junos/junos.py [hunk 1]
```diff
@@ -331,6 +331,11 @@ def get_environment(self):
         for sensor_object, object_data in environment.items():
             structured_object_data = {k: v for k, v in object_data}
 
+            if structured_object_data['class']:
```
Comment (dbarrosop) [inline napalm_junos/junos.py:335]: You can rewrite this as:

```
structured_object_data['class'] = structured_object_data['class'] or current_class
```
or even:

```
structured_object_data['class'] = structured_object_data.get('class', current_class)
```

**Comment by dbarrosop** [comment]
Would you mind adding tests? Just add an extra folder here with the relevant mocked data:

https://github.com/napalm-automation/napalm-junos/tree/develop/test/unit/mocked_data/test_get_environment

**Review by dbarrosop — CHANGES_REQUESTED** [review]
Overall it looks good and doesn't seem to break anything. Would be nice to add tests and make that tiny change in the code I suggested.

File: napalm_junos/junos.py [hunk 2]
```diff
@@ -331,6 +331,11 @@ def get_environment(self):
...
+                current_class = structured_object_data['class']
+            else:
+                structured_object_data['class'] = current_class
```
Comment (mirceaulinic) [inline napalm_junos/junos.py]: Where does `current_class` come from -- doesn't seem to be defined before this test block?

**Comment by bkeifer** [comment]
Getting sucked into a meeting.  Will take care of the build errors later.

File: napalm_junos/junos.py [hunk 3]
```diff
@@ -331,6 +331,8 @@ def get_environment(self):
         for sensor_object, object_data in environment.items():
             structured_object_data = {k: v for k, v in object_data}
 
+            structured_object_data['class'] = structured_object_data['class'] or current_class
```
Comment (mirceaulinic) [inline napalm_junos/junos.py]: Same question as earlier... where is `current_class` defined?

**Comment by mirceaulinic** [comment]
> Getting sucked into a meeting.

Have fun @bkeifer

File: napalm_junos/junos.py (hunk 3 above)
Comment (dbarrosop) [reply napalm_junos/junos.py]: Yeah, now with tests we are getting a CI error complaining that attribute is nit defined.

https://travis-ci.org/napalm-automation/napalm-junos/jobs/257353943#L406

**Comment by bkeifer** [comment]
I'm back to the original conditional logic.  Here's why:

This line, suggested by @dbarrosop _will_ update the object with the correct class name.
`structured_object_data['class'] = structured_object_data['class'] or current_class `

However, since Juniper only gives us a 'class' for the first instance of an object of that class, we also need to track when we see a new class name to let us know that the following data is about a different class of object.  That's where the `else:` and current_class comes in.

I've added some comments to hopefully explain whats going on a bit better, and Travis seems happier now.

File: napalm_junos/junos.py (hunk 1 above)
Comment (dbarrosop) [reply napalm_junos/junos.py:335]: You can write this as:

```
current_class = structured_object_data['class'] or current_class
structured_object_data['class'] = current_class
```
Actually, you don't even need the second statement. You could just change the occurrences of `structured_object_data['class']` with `current_class` so the code would start like:

```
current_class = structured_object_data['class'] or current_class
if current_class == 'Power':
...
```

Comment (dbarrosop) [reply napalm_junos/junos.py:335]: By the way, let me know if I am being nitpicky :P

**Review by mirceaulinic — APPROVED** [review]
Looks good.
Thanks @bkeifer

**Pull request merged**